samza-log4j/src/main/java/org/apache/samza/logging/log4j/StreamAppender.java

```diff
@@ -93,6 +97,17 @@ public void setStreamName(String streamName) {
     this.streamName = streamName;
   }
 
+  public int getPartitionCount() {
+    if (partitionCount > 0) {
+      return partitionCount;
+    }
+    return new JobConfig(getConfig()).getContainerCount();
+  }
+
+  public void setPartitionCount(int partitionCount) {
```

Comment:
Can this be passed in as part of the constructor or the setupSystem method? Not a big fan of order dependent state mutations.

Also, where/how is this set to the user specified value? Does this happen automagically if the field name matches the xml property name?

Reply:
For log4j appenders, when you set a property in the log4j.xml file, it uses the set<property name> and get<property name> as getter/setter automagically.